source/common/poolable_impl.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "nighthawk/common/poolable.h"
+
+#include "common/common/assert.h"
+
+namespace Nighthawk {
+
+class PoolableImpl : public Poolable {
```

Comment:
I think we try to avoid mix-in (multiply inherited) classes like this, generally. Could it just be some kind of `Handle` class that is contained rather than inherited?

Reply:
Let me think about that, will revisit this comment once I did so